gnnboundary/utils/boundary_baseline.py

```diff
@@ -0,0 +1,70 @@
+import random
+import numpy as np
+import networkx as nx
+
+
+class BaselineGenerator:
+
+    def __init__(self, dataset, class_pairs, seed=12345):
+        """
+
+        :param dataset: list of dataset split by predicted class
+        :param class_pairs: list of class idx pair
+        :param seed: seed
+        """
+        assert len(class_pairs) == 2
+
+        random.seed(seed)
+        np.random.seed(42)
+
+        self.dataset_1 = dataset[class_pairs[0]]
+        self.dataset_2 = dataset[class_pairs[1]]
+
+    def sample(self, sample_size):
+        """
+
+        :param sample_size: int of sample size to be generated
+        :return: samples: list of nx Graphs
+        """
+
+        samples = []
+
+        rand_idx_1 = np.random.randint(len(self.dataset_1), size=sample_size)
+        rand_idx_2 = np.random.randint(len(self.dataset_2), size=sample_size)
+
+        for idx_1, idx_2 in zip(rand_idx_1, rand_idx_2):
+            G1 = self.convert_to_nx_graph(self.dataset_1[idx_1])
+            G2 = self.convert_to_nx_graph(self.dataset_2[idx_2])
+
+            #relabel overlapping nodes from G1 and G2
+            G2 = nx.relabel_nodes(G2, lambda x: x + len(G1.nodes))
+
+            #combine G1 and G2 together
+            G = nx.compose(G1, G2)
+
+            #choose 2 random node from each graph
+            node_1 = random.choice(list(G1.nodes))
+            node_2 = random.choice(list(G2.nodes))
+
+            #join by connecting edge
+            G.add_edge(node_1, node_2)
```

Comment:
we should also add the edge from node_2 to node_1 when dealing with an undirected graph

Reply:
Nvm, I checked and since we are using the Graph class and not DiGraph there is no need to add both.